src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -224,264 +283,248 @@ class ReimbursementAccountPage extends React.Component {
-        const subStep = achData.subStep;
-        const shouldShowOnfido = this.props.onfidoToken && !achData.isOnfidoSetupComplete;
-        const backTo = lodashGet(this.props.route.params, 'backTo', ROUTES.HOME);
+        if (!hasACHDataBeenLoaded) {
+            if (reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps && !reimbursementAccount.isLoading) {
+                setShouldShowContinueSetupButton(getShouldShowContinueSetupButtonInitialValue());
+                setHasACHDataBeenLoaded(true);
+            }
+            return;
+        }
+
+        if (
+            prevProps.current.reimbursementAccount.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE &&
+            reimbursementAccount.pendingAction !== prevProps.current.reimbursementAccount.pendingAction
+        ) {
+            setShouldShowContinueSetupButton(hasInProgressVBBA());
+        }
+
+        const currentStep = lodashGet(reimbursementAccount, 'achData.currentStep') || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+
+        if (shouldShowContinueSetupButton) {
+            return;
+        }
+
+        const currentStepRouteParam = getStepToOpenFromRouteParams();
+        if (currentStepRouteParam === currentStep) {
+            return;
+        }
+        if (currentStepRouteParam !== '') {
+            BankAccounts.hideBankAccountErrors();
+        }
+
+        const backTo = lodashGet(route.params, 'backTo');
+        const policyId = lodashGet(route.params, 'policyID');
+        Navigation.navigate(ROUTES.BANK_ACCOUNT_WITH_STEP_TO_OPEN.getRoute(getRouteForCurrentStep(currentStep), policyId, backTo));
+
+        // At the end, update the prevProps ref to the current values
+        prevProps.current = {
+            reimbursementAccount,
+            network,
+            route,
+            onfidoToken,
+            policy,
+            account,
+            isLoadingReportData,
+            session,
+            translate,
+            plaidLinkToken,
+            reimbursementAccountDraft,
+        };
+    }, [reimbursementAccount, network, route, hasACHDataBeenLoaded, shouldShowContinueSetupButton]);
+
+    /**
+     * @param {String} currentStep
+     * @returns {String}
+     */
+    function getRouteForCurrentStep(currentStep) {
         switch (currentStep) {
-            case CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT:
-                if (this.hasInProgressVBBA()) {
-                    this.setState({shouldShowContinueSetupButton: true});
-                }
-                if (subStep) {
-                    BankAccounts.setBankAccountSubStep(null);
-                } else {
-                    Navigation.goBack(backTo);
-                }
-                break;
             case CONST.BANK_ACCOUNT.STEP.COMPANY:
-                BankAccounts.goToWithdrawalAccountSetupStep(CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, {subStep: CONST.BANK_ACCOUNT.SUBSTEP.MANUAL});
-                break;
+                return 'company';
```

Comment:
I assume we can create constants for these values: `company`, `personal-information`....